pkg/controller/psbackup/controller_test.go

```diff
@@ -406,6 +407,116 @@ func TestRunningState(t *testing.T) {
 	}
 }
 
+func TestGetBackupSource(t *testing.T) {
```

Comment:
Let's keep in mind that this test will require some refactor in the future given that the existing functionality before the changes of this pr are not tested at all.